src/Symfony/Component/Process/Tests/ProcessTest.php

```diff
@@ -1427,9 +1428,10 @@ public function testInheritEnvDisabled()
      *
      * @return Process
      */
-    private function getProcess($commandline, $cwd = null, array $env = null, $input = null, $timeout = 60, array $options = array())
+    private function getProcess($commandline, $cwd = null, array $env = null, $input = null, $timeout = 60)
     {
-        $process = new Process($commandline, $cwd, $env, $input, $timeout, $options);
+        $process = new Process($commandline, $cwd, $env, $input, $timeout);
+        $process->inheritEnvironmentVariables(true, false);
```

Comment:
> I don't think this is the best upgrade path.
> we should give them the right update path from the start IMO.

That's true - a smoother upgrade path would be to not deprecate the method in 3.x, but do it in 4.1. We already did so in the past.
Yet, I think this is not worth it for the feature we're talking about. So a more brutal - yet continuous - upgrade path in enough here. At least that what I'm proposing here.

Taking this into account, what path to you think we should take?